Interfaces/API/Job.py

```diff
@@ -578,39 +579,33 @@ def setDestination( self, destination ):
     """
     kwargs = {'destination':destination}
     if isinstance( destination, basestring ):
-      if not re.search( '^DIRAC.', destination ) and not destination.lower() == 'any':
-        result = self.__checkSiteIsValid( destination )
+      destination = [destination]
+      description = 'User specified destination site'
+    else:
+      description = 'List of sites selected by user'
+    if isinstance( destination, list ):
+      sites = set( site for site in destination if not re.search( '^DIRAC.', site ) and site.lower() != 'any' )
```

Comment:
No, I have no idea why this was done and I didn't dare changing it... Probably at the time since getSiteCEMapping was used, the author (Stuart ;-) was not sure DIRAC sites would declare a CE in the CS... As now the method getSites just gets the list of sites, we could suppress this check.
If you agree, I will remove it (but keep the check on "any" of course ;-)